subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultIvyArtifactRepository.java

```diff
@@ -367,6 +367,17 @@ protected void invalidateDescriptor() {
         schemes = null;
     }
 
+    public boolean hasStandardPattern() {
+        // This is wasteful because we create a descriptor and throw it away immediately.
+        IvyRepositoryDescriptor descriptor = createDescriptor();
```

Comment:
Does this problem manifest in Maven, too? Which also extends `AbstractResolutionAwareArtifatRepostory`? 

Could we just update the base class to avoid caching?

Reply:
This is Ivy only

Reply:
Unfortunately, it would need to be the other way around.

We call `hasStandardPattern` before task execution, which causes the descriptor to be cached. Later, during task execution, the URL for the repository changes and the descriptor needs to be invalidated.

So instead of caching the descriptor before task execution and having to invalidate it later, we just look at a new descriptor each time.